src/sources/soundsourceffmpeg.cpp

```diff
@@ -1205,93 +1179,190 @@ ReadableSampleFrames SoundSourceFFmpeg::readSampleFramesClamped(
-                const auto missingRange = IndexRange::between(
-                        writableFrameRange.start(),
-                        readFrameIndex);
-                const auto clearFrameRange = intersect(missingRange, writableFrameRange);
-                if (clearFrameRange.length() > 0) {
+                const auto missingFrameRange =
+                        IndexRange::between(
+                                writableFrameRange.start(),
+                                math_min(writableFrameRange.end(), readFrameIndex));
+                if (!missingFrameRange.empty()) {
+                    kLogger.warning()
+                            << "Generating silence for missing sample data"
+                            << missingFrameRange;
+                    const auto clearFrameCount =
+                            missingFrameRange.length();
                     const auto clearSampleCount =
-                            getSignalInfo().frames2samples(clearFrameRange.length());
+                            getSignalInfo().frames2samples(missingFrameRange.length());
                     if (pOutputSampleBuffer) {
                         SampleUtil::clear(
                                 pOutputSampleBuffer,
                                 clearSampleCount);
                         pOutputSampleBuffer += clearSampleCount;
                     }
-                    writableFrameRange.shrinkFront(clearFrameRange.length());
+                    writableFrameRange.shrinkFront(clearFrameCount);
+                }
+            }
+            DEBUG_ASSERT(writableFrameRange.empty() ||
+                    writableFrameRange.start() >= readFrameIndex);
+
+            // -= 2nd step =-
+            // Check for skipped sample data and log
+            DEBUG_ASSERT(readFrameIndex <= decodedFrameRange.start());
+            const auto skippedFrameRange =
+                    IndexRange::between(
+                            readFrameIndex,
+                            decodedFrameRange.start());
+            if (!skippedFrameRange.empty()) {
+                // The decoder has skipped some sample data that needs to
+                // be filled with silence to continue decoding! This is supposed
+                // to occur only at the beginning of a stream for the very first
+                // decoded frame with a lead-in due to start_time > 0. But not all
+                // encoded streams seem to account for this by correctly setting
+                // the start_time property.
+                // NOTE: Decoding might even start at a negative position for the
+                // first frame of the file, i.e. outside of the track's valid range!
+                // Consequently isValidFrameIndex(readFrameIndex) might return false.
+                // This is expected behavior and will be compensated during 'preskip'
+                // (see below).
+                if (readFrameIndex <= frameIndexRange().start()) {
```

Comment:
This is hard to understand because nothing happens here.
Can we either copy the code here and accept duplication or reference the code below better by comments.

Reply:
No, because `readFrameIndex` will be updated during the following steps. Printing the log for the whole `skippedFrameRange` when available and current seems more reasonable to me. I'll clarify the comments.